superset/charts/api.py

```diff
@@ -561,9 +561,13 @@ def data(self) -> Response:
             command = ChartDataCommand()
             query_context = command.set_query_context(json_body)
             command.validate()
+        except QueryObjectValidationError as error:
+            return self.response_400(message=error.message)
         except ValidationError as error:
             return self.response_400(
-                message=_("Request is incorrect: %(error)s", error=error.messages)
+                message=_(
+                    "Request is incorrect: %(error)s", error=error.normalized_messages()
+                )
```

Comment:
Use `normalized_messsages` from mashmallow's `ValidationError` so that the error message shows which query field failed the validation.